### ISSUE TITLE
Board can't boot

### ISSUE THREAD

**Issue by jwrz** [avafinger/bpi-m64-firmware#3]
Hi, I prepared SD card according to your instruction (with problems becouse files from gdrive can't be wgeted but i did it manualy). When I try to boot my BPI M64 with this card I obtain black screen.. nothing happend. I repeat whole process but effect is still the same. Maybe you can publish image file with everything prepared?

**Comment by avafinger** [comment]
Black screen could mean three things:

a) You Monitor is not 1080p capable, so you will see a black screen.
To find out it is this problem, please, boot the board with SD card and wait for a minute or so and just type on keyboard: ubuntu and look at the ethernet if you see some activities. You must have a DHCP server to release an IP to the board. Default is DHCP.

b)  HDMI to DVI, there are some information that need to get pass through it, depending on your converter you may find trouble.

c)  SD card corrupted or you SDHC card reader sucks, try to get another SD card or use another SD card reader.

I will push a DTB for 720p that could help.

**Comment by jwrz** [comment]
Thank you for replay. I have new observations. 
My monitor is capable of 1080p - when bpi is trying to boot, the monitor is showing text "1080p input." There is no HDMI-DVI adapters. I was trying to use 5 cards (GOODRAM 16GB SDHC toschiba 8G SDHC x2, SanDisk SD 4G x2) and 3 readers (3 laptops - diferent vendors too).

BPI try to boot a 3 or 4 times and turn off. While booting I saw white lines on screen, sometimes 4 tuxes.

**Comment by avafinger** [comment]
You should see 4 tuxes and then boot messages.
"... and turn off" is an indication your PSU is under specifications, you need 5v and ~2.5A.
The board i have comes with a barrel connector (DC input), are you using the barrel connector?

Try to boot without anything attached to the USB ports, just to see if it changes something.

After 3 or 4 attempts to boot without success your SD card may be corrupted, see: c. Check for SD card integrity

**Comment by jwrz** [comment]
I try Anker 60W 6x2.4A (USB) and lab power supply 10A (DC input) - I also checked it with oscilloscope (while booting) and it looks good.
I have two m64 boards - both have the same problem.
I checked SD integrity - you will not suprise, they are good.


Are there any logs created while booting, which I can check?

**Comment by avafinger** [comment]
Yes, you can grab the boot log with a TTL serial debug, please do so and then we can try to see what is going on with the board.

attach the complete boot log here or in gist if you prefer.

You did not answer:
"The board i have comes with a barrel connector (DC input), are you using the barrel connector?"

**Comment by jwrz** [comment]
OK, I will grab it later. 
According to your question: ...lab power supply 10A (barrel connector - DC input)...

**Comment by jsoncylu** [comment]
I am also having the same issue after creating the SD card. I can't get any output via TTL using a baudrate of 115200. I am going to try using a different SD card reader.

**Comment by avafinger** [comment]
@setup: 115200 8-N-1

TX -> RX
RX -> TX
GND -> GND

**Comment by avafinger** [comment]
For those of you having trouble i would suggest using SanDisk Ultra 16 GB (or greater) 48 MB/s or 80 MB/s, i have several cards and have never run into trouble with Ultra. They are cheap, slow but works and save you time....

**Comment by jsoncylu** [comment]
Yes, I am using the one you suggested, and I used Ubuntu to format the card. I was using my laptop's built in SD card reader but that might be crappy.

**Comment by avafinger** [comment]
Does your board powers off also, or just black screen?

**Comment by jsoncylu** [comment]
I think just black screen. No tuxes. The LED stays red.

**Comment by avafinger** [comment]
Normal boot the LED stays red all the time, try the 720p DTB and see what happens.
Mind the the HDMI connector has some trouble, sometimes i have to press against the board and the image pops up on screen. But i see Ethernet activities when the board gets the IP.
Try to type the password: ubuntu after 30 / 50 secs and see if there is some eth0 activities.

**Comment by jwrz** [comment]
Hi, this is log file grabbed from serial port
[log.txt](https://github.com/avafinger/bpi-m64-firmware/files/797153/log.txt)

**Comment by avafinger** [comment]
Hmm, ok. The board dropped to a u-boot command line, for whatever reason it is sensing ENTER on serial console debug and drops to u-boot shell.
I can see by the log several ENTERs.
Usually it would drop to shell and you would be able to type commands, type in the command:
sunxi# run mmcboot [ENTER]

** Update: run mmcbootcmd

This will boot and load kernel as normal, but you have to find out why serial console (debug) gets [ENTER] if you don't have anything attached to serial console.
I sometimes get this situation on Pine64 but not with the ENTERs you have, and i just type: run mmcboot and tada!

If you can't type anything, try to reboot until it finally boots or try to boot without anything attached to USB with the TTL attached and without it.

**Comment by jwrz** [comment]
I send this command by serial - "echo -ne 'run mmcbootcmd\r' > /dev/ttyAMA0 | cat /dev/ttyAMA0 > /home/jwrz/log4.txt".

[log4.txt](https://github.com/avafinger/bpi-m64-firmware/files/797745/log4.txt)

**Comment by avafinger** [comment]
Almost there!

For some reason the board rebooted. I think you should review how you are powering the board or placing the board on a static surface.
Here is the complete kernel boot log booting from eMMC:
https://github.com/avafinger/bpi-m64-firmware/blob/master/img/dmesg_kernel_3.10.104.txt

I can see you seems to be a knowledge person or works in the field, you have all the stuff according to the specifications, PSU with 10A should be enough (maybe undervolt?), for instance do you have a second bpi-m64?

I am running out of ideas what you could try..... Have you flashed Android? Did it Work?
Have you run any other image from SD CARD?

*Update: Maybe you are facing similar problem described here: https://forum.armbian.com/index.php/topic/3169-request-for-banana-pi-m64/?p=26306

**Comment by jwrz** [comment]
I will be trying. I will tell you, if I find the reason.

I have two M64 and both have the same problem.
Image  from banana-pi.org/m64-download.html (raspbian) with your DTB (https://github.com/avafinger/dtb_bpi-m64/pine64-plus.dtb not sun50i-a64-pine64-plus.dtb) works well, but without the mmc.

**Comment by avafinger** [comment]
I suspect there is two different batch of boards released. sounds like the culprit is u-boot in this case.

You could try this:
- Write the (raspian) u-boot or the pine64 u-boot to your SD CARD
- Create the partition the new u-boot  requires (the one you have chosen above)  and write the files in the partition and with the correct names

or

On the raspian SD CARD just update the kernel as described on section ** Updating Kernel to 3.10.104 **

If one of this works we know the problem is u-boot, in this case you should stick to original u-boot.

**Comment by avafinger** [comment]
@jwrz

Please, try this on the SD CARD image (from your HOST PC) and report back if anything changed:

```
mkdir -p m64_test
cd m64_test
wget https://github.com/avafinger/a64_bin/raw/master/ub-m64-sdcard-test.bin
sudo dd if=./ub-m64-sdcard-test.bin conv=notrunc bs=1k seek=19096 of=${SDCARD}
sync
```

where 
`${SDCARD} is you SD CARD like /dev/sdb`

and umount your SD CARD after 'dd'

**Comment by jwrz** [comment]
Hi,
I did it and... it works :) but to boot linux, it was needed to use 'echo -ne 'run mmcbootcmd\r' > /dev/ttyAMA0' again.

**Comment by jodyono** [comment]
Are you able to boot and run with the LCD touchscreen?  I'm able to boot and run with the updated kernel if I only use HDMI.  When I switch the device tree to use the LCD and install the touch drivers, it won't boot at all.

**Comment by avafinger** [comment]
Please, provide more info about your LCD, model, how you did install it.
Are you able to grab the boot log? Can you see if you type: ubuntu after
30/50 secs you see eth0 activities?

I don't have the LCD but it should work...

**Comment by jodyono** [comment]
I have the [official Banana Pi LCD touchscreen](https://www.amazon.com/Banana-Pi-Official-Touchscreen-Display/dp/B01BM5XARQ) and it works with the BPI-M64 Android 6.0 LCD build.  I'm going to make my Linux SD card again from scratch and then try what you have suggested.  I'll let you know if I can get the boot log and/or see any Ethernet activity.

**Comment by avafinger** [comment]
Can you tell if LCD gets the backlight ON or turns ON with the new DTB for
the LCD?

**Comment by jodyono** [comment]
In my last attempt, NOTHING turned-on.

**Comment by avafinger** [comment]
Ok, i will check the DTB again, maybe you need the u-boot with the LCD
changes also. will see what i can do.

**Comment by jodyono** [comment]
Many thanks!

**Comment by avafinger** [comment]
please, try with this and report back if it works:
https://github.com/avafinger/bpi-m64-firmware/raw/master/a64-2GB_LCD_TOUCH_OK.dtb

**Comment by avafinger** [comment]
@jodyono ,

Any news about the Touch?

**Comment by jodyono** [comment]
Still not working, but I'm not sure what's wrong.  My system no longer boots at all, i.e., there's no boot.log.  I might have made a mistake while creating the SD card.  I have to admit to being a little confused by the instructions, especially at the point where I'm supposed to add TSLIB support.  Is that absolutely necessary just to see if the display lights up?

**Comment by avafinger** [comment]
TSLIB is just for the touch, not needed to turn the LCD on.
The latest DTB has a  fix to activate LCD that is or was supposed to work!
Just make sure you renamed the new one (a64-2GB_LCD_TOUCH_OK.dtb
<https://github.com/avafinger/bpi-m64-firmware/blob/master/a64-2GB_LCD_TOUCH_OK.dtb>
) to a64-2GB.dtb.
Please, if after you rename the file the board does not boot i messed
something, restore the previous one to boot again.
Sorry if does not work.

**Comment by avafinger** [comment]
@jodyono ,
The reason the LCD panel did not work is due to missing driver in the kernel.
I adding the driver this weekend, can you please test if this works?

**Comment by jodyono** [comment]
I can test it, but not until I get back from a business trip in a couple of weeks.

**Comment by avafinger** [comment]
@jodyono ,
When you get back, please give it a try and see if works. I think it will work...

**Comment by avafinger** [comment]
@jodyono 
If you do test, please try with this dtb: https://github.com/avafinger/bpi-m64-firmware-v2/blob/master/a64-2GB_LCD7-v4.dtb

Just overwrite the a64-2GB.dtb (boot partition) with this a64-2GB_LCD7-v4.dtb.
I think we may need a special u-boot with this settings, but lets try without one for now. And use the kernel from: https://github.com/avafinger/bpi-m64-firmware-v2 with latest LCD driver.

**Comment by avafinger** [comment]
Someone tested and 7" LCD works: https://github.com/avafinger/bpi-m64-firmware-v2#7-lcd-support

Use this Image for latest kernel and LCD support.
